packages/proxy/lib/http/util/cookies.ts

```diff
@@ -10,6 +10,47 @@ interface RequestDetails {
   needsCrossOriginHandling: boolean
 }
 
+/**
+ * Whether or not a url's scheme, domain, and top-level domain match to determine whether or not
+ * a cookie is considered first-party. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#third-party_cookies
+ * for more details.
+ * @param {string} url1 - the first url
+ * @param {string} url2 - the second url
+ * @returns {boolean} whether or not the URL Scheme, Domain, and TLD match. This is called same-site and
+ * is allowed to have a different port or subdomain. @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site#directives
+ * for more details.
+ */
+export const areUrlsSameSite = (url1: string, url2: string) => {
+  if (!url1 || !url2) return false
+
+  const { port: port1, ...parsedUrl1 } = cors.parseUrlIntoDomainTldPort(url1)
+  const { port: port2, ...parsedUrl2 } = cors.parseUrlIntoDomainTldPort(url2)
+
+  // If HTTPS, ports NEED to match. Otherwise, HTTP ports can be different and are same origin
+  const doPortsPassSameSchemeCheck = port1 !== port2 ? (port1 !== '443' && port2 !== '443') : true
```

Comment:
noob question - what does 443 represent?

Reply:
default HTTPS port. Only reason I am adding the check here is due to the logic into `parseUrlIntoDomainTldPort`. I am wondering for checking same-site capability if we want to go off the protocol scheme as well, since `443` might be SSL, and there could be other ports that are https.

For example, `https://www.testme.com:8443` and (`https://app.testme.com:4849`|`http://app.testme.com:443`) are all same-site, but this function would evaluate `false` | `true`. Since we infer port from`parseUrlIntoDomainTldPort`, I am not sure this is an issue though